File: monitor/bin/js/requests-by-ip.js
const { readLines } = require('ez-tools');
const { debuglog } = require('util');
const debuglogFunctions = debuglog('functions');

/*
  Print a lists of requests from a given IP
*/

async function requestsByIp(ip) {
  debuglogFunctions('requestsByIp');
  let logLines = await readLines('/var/log/nginx/access.log');
  let ipLines = logLines.filter(checkForIp);
  ipLines.forEach(line => console.log(line));

  function checkForIp(line) {
    return line.match(ip);
  }
}

if (!process.argv[2]) {
  console.error('usage: requests-by-ip.js [ip]');
  process.exit(1);
}

requestsByIp(process.argv[2])
  .catch(function(err) {
    console.error(err);
  });

File: monitor/bin/rb/helpers/requests_by_ip.rb
# Returns an array of all requests from an ip

def requests_by_ip(ip)
  ip_lines = []
  File.open("#{ENV['NGINX_LOG_PATH']}/access.log").each do |line|
    if line.match(ip) then ip_lines.push(line) end
  end
  ip_lines
end
